graylog2-server/src/test/java/org/graylog/plugins/views/search/searchtypes/pivot/buckets/TimeUnitIntervalTest.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.plugins.views.search.searchtypes.pivot.buckets;
+
+import org.elasticsearch.search.aggregations.bucket.histogram.DateHistogramInterval;
+import org.graylog2.plugin.indexer.searches.timeranges.InvalidRangeParametersException;
+import org.graylog2.plugin.indexer.searches.timeranges.RelativeRange;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+public class TimeUnitIntervalTest {
+    private TimeUnitInterval.Builder builder() {
+        return TimeUnitInterval.Builder.builder();
+    }
+
+    @Test
+    public void doesNotAllowInvalidTimeUnit() {
+        assertThatExceptionOfType(RuntimeException.class)
+                .isThrownBy(() -> builder().timeunit("foobar").build())
+                .withMessage("Time unit must be {quantity}{unit}, where quantity is a positive number and unit [smhdwM].");
```

Comment:
:+1: for validating the exception message! I usually put the message in a static field on the class and reference it in the test to avoid having it to change it in two places when the time comes.

Reply:
I do understand that, but I usually prefer to have literals in tests, because it is harder for me to overlook mismatching expectations because I _think_ I know what the variable is.